packages/mask/background/database/avatar-cache/avatar.ts

```diff
@@ -9,29 +8,44 @@ import { createTransaction } from '../utils/openDB'
  * Get a (cached) blob url for an identifier. No cache for native api.
  * ? Because of cross-origin restrictions, we cannot use blob url here. sad :(
  */
-export const queryAvatarDataURL = (
-    hasNativeAPI
-        ? async function (identifier: IdentifierWithAvatar): Promise<string | undefined> {
-              return nativeAPI?.api.query_avatar({ identifier: identifier.toText() })
-          }
-        : memoizePromise(
-              async function (identifier: IdentifierWithAvatar): Promise<string | undefined> {
-                  const t = createTransaction(await createAvatarDBAccess(), 'readonly')('avatars')
-                  const buffer = await queryAvatarDB(t, identifier)
-                  if (!buffer) throw new Error('Avatar not found')
-                  return blobToDataURL(new Blob([buffer], { type: 'image/png' }))
-              },
-              (id) => id.toText(),
-          )
-) as ((identifier: IdentifierWithAvatar) => Promise<string | undefined>) & {
-    cache?: Map<string, unknown>
+async function nativeImpl(identifiers: IdentifierWithAvatar[]): Promise<IdentifierMap<IdentifierWithAvatar, string>> {
+    const map = new IdentifierMap<IdentifierWithAvatar, string>(new Map())
+    await Promise.allSettled(
+        identifiers.map(async (id) => {
+            const result = await nativeAPI!.api.query_avatar({ identifier: id.toText() })
```

Comment:
```suggestion
            const result = await nativeAPI?.api.query_avatar({ identifier: id.toText() })
```

Reply:
It's an impossible case because we checked before. If it really happens, it should throw.